src/diffusers/loaders/single_file_utils.py

```diff
@@ -305,7 +307,11 @@ def is_valid_url(url):
     return original_config
 
 
-def infer_model_type(original_config, model_type=None):
+def infer_model_type(original_config, checkpoint=None, model_type=None):
+    if checkpoint is not None:
```

Comment:
Since checkpoint is always passed I think we can move it to below the `model_type` check. This way if `model_type` gets set correctly if explicitly passed.

Reply:
Should be done in 2be231cc.